### PR TITLE
Enhance style rules

### PR DESCRIPTION
**Pull request by apettenati**
Adds a few style rules to enforce consistent styling for a few common differences.

Most of these are auto-fixable, so this should allow your formater to handle these cases automatically.

Opened an issue to demonstrate what these auto-fixes look like. Let me know your thoughts and if we want to change any of these!
https://github.com/TechChange/course-builder/pull/1001

### CODE REVIEW

File: rules/style.js [hunk 1]
```diff
@@ -135,10 +158,13 @@ module.exports = {
...
 		// Require space around infix operators (e.g. var foo = 1 + 2, not var foo = 1+2)
 		"space-infix-ops": 2,
-		// Require spaces around unary words like void, new, delete, disallow around nonwords like ++, --, !
+		// Require spaces around unary words like void, new, delete,
+		// disallow around nonwords like ++, --, !
 		"space-unary-ops": 2,
 		// Require spaces in comments (e.g. // This is a comment, not //This is a comment)
 		"spaced-comment": [2, "always"],
+		// Never allow spaces within the curly braces of template strings
+		"template-curly-spacing": [2, "never"],
```
Comment (apettenati) [inline rules/style.js:167]: For the most part we have no spaces here, but there are a few updates. Will auto-format!

File: rules/style.js [hunk 2]
```diff
@@ -1,14 +1,20 @@
 module.exports = {
 	"rules": {
-		// Enforce no spaces at bookends of array/object definitions (e.g. [2, 3] is valid, [ 2, 3 ] is invalid)
+		// Enforce adding line breaks consistently. Reports an error if one
+		// bracket in the pair has a linebreak and the other does not.
+		"array-bracket-newline": [2, "consistent"],
```
Comment (apettenati) [inline rules/style.js:5]: No violations, but should improve auto-formatting!

File: rules/style.js [hunk 3]
```diff
@@ -107,9 +123,16 @@ module.exports = {
 		}],
 		// Require extra spaces inside objects
 		"object-curly-spacing": [2, "always"],
-		// Require quoting of object keys when necessary (e.g. var object = { "foo-bar": 5, bar: 2, foo: function() {}})
+		// Enforce operators at the end of a line except for ternary operators,
+		// which should be at the beginning of a line
+		"operator-linebreak": [2, "after",
+			{ "overrides": {"?": "before", ":": "before"}}
+		],
```
Comment (apettenati) [inline rules/style.js:130]: I think we've agreed we want to put `&&` and `||` at the end of lines and ternary operators at the beginning, correct? Happy to change to our preference. This is what drives most of the changes.

Comment (apettenati) [reply rules/style.js:130]: This will auto-format!

**Review by jantonso — APPROVED** [review]
Amazing!

**Review by beckybell13 — APPROVED** [review]
Thank you!!

**Pull request merged**